src/program.ts

```diff
@@ -944,6 +944,34 @@ export class Program extends DiagnosticEmitter {
       : null;
   }
 
+  /** Replace a old source with a new source according to internalPath */
+  replaceSource(newSource: Source): void {
+    var parser = this.parser;
+    parser.donelog.delete(newSource.internalPath);
+    parser.seenlog.delete(newSource.internalPath);
```

Comment:
Is the `delete` meaningful ? I found that near-sdk will do such an operation, but I'm not sure it's neccessary.

Reply:
cc @willemneal

Reply:
At first glance such delete and add same paths looks unnecessary